### PR TITLE
Remove order reads from API

### PR DESCRIPTION
**Pull request by mbj**
Extraction from #398

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review as we deployed a replacement for this endpoint to our main app.

**Comment by dkubb** [comment]
@mbj are there any views or `#as_json`, `#to_json` methods related to this that should also be removed?

**Comment by mbj** [comment]
> @mbj are there any views or #as_json, #to_json methods related to this that should also be removed?

Yes, views. Good catch. Will include them.

File: api/config/routes.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@
   namespace :api, defaults: { format: 'json' } do
     resources :option_types, only: :index
 
-    resources :orders, only: %i[index show] do
+    resources :orders, only: %i[] do
```
Comment (dkubb) [inline api/config/routes.rb]: Can this be changed to `namespace :orders` without breaking the routing?

**Comment by dkubb** [comment]
OT but should we be using rabl for the views on the mrh-spree side of things? It seems odd to have all that code within the controllers.

**Comment by mbj** [comment]
> OT but should we be using rabl for the views on the mrh-spree side of things? It seems odd to have all that code within the controllers.

No, we explicitly discussed this in the past and resulted in: Wanting to get a rid of rabl:

* The code should be mutation testable
* Rabls is kinda slow and needs caching behavior, that is complexity in itself
* Its implementation is terrible

As the main app side of things was an 1:1 port to keep the client unchanged the JSON we have to produce is very large. But we can reduce this once we can do an iteration to reduce the client, and add support for batches. This will remove the remaining unpleasantness in the mrh side implementation.

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

File: api/config/routes.rb (hunk 1 above)
Comment (mbj) [inline api/config/routes.rb]: Apparently not as we do need to generate the order collection member params, which `namespace` would not do.

**Pull request merged**